### PR TITLE
Adding testcase to passthrough the multifunction adaptor to guest

### PR DESCRIPTION
**Pull request by prudmirya**
This script is for pass through the multifunction adaptor to the guest and guest can directly use it.   
Signed-off-by: prudhvi <mprudhvi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by prudmirya** [comment]
above test case passthrough the NIC adaptor

File: libvirt/tests/cfg/libvirt_pci_passthrough_multifunction.cfg [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            # device to guest. Then this network device
+            # will be unavailable on host.
+            # E.g: 0000:05:00.0
+            libvirt_pci_net_dev_label = ""
+            # please enter Company name to check inside the guest
+            libvirt_pci_net_dev_name = "Broadcom"
+            # Please enter the ip what is used by the device
+            # you are going to attach to guest.
+            libvirt_pci_net_ip = ""
+            # Please enter a vailable ip from the net device.
```
Comment (nasastry) [inline libvirt/tests/cfg/libvirt_pci_passthrough_multifunction.cfg]: s/a vailable/available/

File: libvirt/tests/cfg/libvirt_pci_passthrough_multifunction.cfg [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            # We need to ping it after attaching pci device
+            # to guest to verify this device works well in guest.
+            libvirt_pci_server_ip = ""
+            # enter the CIDR notation
```
Comment (nasastry) [inline libvirt/tests/cfg/libvirt_pci_passthrough_multifunction.cfg]: s/enter the CIDR notation/Enter netmask in CIDR notation/

File: libvirt/tests/src/libvirt_pci_passthrough_multifunction.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+def run(test, params, env):
+    """
+    Test for PCI multifunction device passthrough to libvirt guest.
+
+    a). NIC:
+        1. Get params.
+        2. Get the pci device function.
+        3. Attach each pci device function to guest.
+        4. Start guest and set the ip to all the functions.
+        5. Ping to server_ip from ecah function
```
Comment (nasastry) [inline libvirt/tests/src/libvirt_pci_passthrough_multifunction.py]: s/ecah/each/

File: libvirt/tests/src/libvirt_pci_passthrough_multifunction.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    netmask = params.get("libvirt_pci_net_mask", "ENTER.YOUR.Mask")
+
+    # Check the parameters from configuration file.
+    if (pci_id.count("ENTER")):
+        raise error.TestNAError("Please enter your device name for test.")
+
+    if (device_type == "NIC" and (net_ip.count("ENTER") or
+                                  server_ip.count("ENTER") or
+                                  netmask.count("ENTER"))):
+        raise error.TestNAError("Please enter the ips for NIC test.")
```
Comment (nasastry) [inline libvirt/tests/src/libvirt_pci_passthrough_multifunction.py]: You need to hint the config file.
some thing like 'Please enter the ips for NIC test in config file'

File: libvirt/tests/src/libvirt_pci_passthrough_multifunction.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                    "No Ethernet found for the pci device in guest.")
+        # check all functions get same bus and slot
+        output = session.cmd_output("lspci -nn | grep %s" % device_name)
+        nic_list = output.splitlines()
+        for i in range(len(nic_list)):
+            bus_info.append(str(nic_list[i]).split(' ', 1)[0])
+            nic_list[i] = str(nic_list[i]).split(' ', 1)[0][:-2]
+        if len(set(nic_list)) != 1:
+            raise error.TestFail(
+                "allocates different slots for same adapter")
```
Comment (nasastry) [inline libvirt/tests/src/libvirt_pci_passthrough_multifunction.py]: Please re-phrase the above sentence.

File: libvirt/tests/src/libvirt_pci_passthrough_multifunction.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
+import os
+import aexpect
+from autotest.client import utils
+from autotest.client.shared import error
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_pci_passthrough_multifunction.py]: please use remove autotest dependencies

File: libvirt/tests/src/libvirt_pci_passthrough_multifunction.py [hunk 7]
```diff
@@ -0,0 +1,105 @@
+import os
+import aexpect
+from autotest.client import utils
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_pci_passthrough_multifunction.py]: `utils` is not used anywhere.

File: libvirt/tests/src/libvirt_pci_passthrough_multifunction.py [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+                raise error.TestFail(
+                    "No Ethernet found for the pci device in guest.")
+        # check all functions get same bus and slot
+        output = session.cmd_output("lspci -nn | grep %s" % device_name)
+        nic_list = output.splitlines()
+        for i in range(len(nic_list)):
+            bus_info.append(str(nic_list[i]).split(' ', 1)[0])
+            nic_list[i] = str(nic_list[i]).split(' ', 1)[0][:-2]
+        if len(set(nic_list)) != 1:
+            raise error.TestFail(
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_pci_passthrough_multifunction.py]: use `test.fail()/test.error()/test.cancel()` in place of `error.TestFail/error.TestError/error.TestNAError`

**Pull request closed without merging**